operator/builtin/transformer/remove/remove.go

```diff
@@ -0,0 +1,82 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package remove
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/open-telemetry/opentelemetry-log-collection/entry"
+	"github.com/open-telemetry/opentelemetry-log-collection/operator"
+	"github.com/open-telemetry/opentelemetry-log-collection/operator/helper"
+)
+
+func init() {
+	operator.Register("remove", func() operator.Builder { return NewRemoveOperatorConfig("") })
+}
+
+// NewRemoveOperatorConfig creates a new restructure operator config with default values
+func NewRemoveOperatorConfig(operatorID string) *RemoveOperatorConfig {
+	return &RemoveOperatorConfig{
+		TransformerConfig: helper.NewTransformerConfig(operatorID, "remove"),
+	}
+}
+
+// RemoveOperatorConfig is the configuration of a restructure operator
+type RemoveOperatorConfig struct {
+	helper.TransformerConfig `mapstructure:",squash" yaml:",inline"`
+
+	Fields []entry.Field `mapstructure:"fields"  json:"fields" yaml:"fields"`
+}
+
+// Build will build a Remove operator from the supplied configuration
+func (c RemoveOperatorConfig) Build(context operator.BuildContext) ([]operator.Operator, error) {
+	transformerOperator, err := c.TransformerConfig.Build(context)
+	if err != nil {
+		return nil, err
+	}
+
+	removeOperator := &RemoveOperator{
+		TransformerOperator: transformerOperator,
+		Fields:              c.Fields,
+	}
+
+	return []operator.Operator{removeOperator}, nil
+}
+
+// RemoveOperator is an operator that deletes a field
+type RemoveOperator struct {
+	helper.TransformerOperator
+	Fields []entry.Field
+}
+
+// Process will process an entry with a restructure transformation.
+func (p *RemoveOperator) Process(ctx context.Context, entry *entry.Entry) error {
+	return p.ProcessWith(ctx, entry, p.Transform)
+}
+
+// Transform will apply the restructure operations to an entry
+func (p *RemoveOperator) Transform(entry *entry.Entry) error {
+	if p.Fields != nil {
+		for _, field := range p.Fields {
+			_, exist := entry.Delete(field)
+			if !exist {
+				return fmt.Errorf("remove: field does not exist")
+			}
+		}
+		return nil
+	}
+	return fmt.Errorf("remove: missing required field 'fields'")
```

Comment:
This check needs to happen in `Build`. We can know whether it is set at that point and error out then.

There are two problems with doing the check here:
1) The user doesn't get immediate feedback about their configuration error
2) The check has to be performed for every single log entry, so becomes an unnecessary performance hit